codex-rs/core/src/client_common.rs

```diff
@@ -12,11 +12,35 @@ use std::pin::Pin;
 use std::task::Context;
 use std::task::Poll;
 use tokio::sync::mpsc;
+use once_cell::sync::Lazy;
+use std::process::Command;
+use std::process::Stdio;
 
 /// The `instructions` field in the payload sent to a model should always start
 /// with this content.
 const BASE_INSTRUCTIONS: &str = include_str!("../prompt.md");
 
+static RG_AVAILABLE: Lazy<bool> = Lazy::new(|| {
+    Command::new("rg")
+        .arg("--version")
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .status()
+        .map(|s| s.success())
+        .unwrap_or(false)
+});
+
+const RG_LINE: &str = "- Do not use `ls -R`, `find`, or `grep` - these are slow in large repos. Use `rg` and `rg --files`.";
+const RG_LINE_NO_RG: &str = "- Do not use `ls -R`, `find`, or `grep` - these are slow in large repos.";
+
+fn base_instructions() -> Cow<'static, str> {
+    if *RG_AVAILABLE {
+        Cow::Borrowed(BASE_INSTRUCTIONS)
+    } else {
+        Cow::Owned(BASE_INSTRUCTIONS.replace(RG_LINE, RG_LINE_NO_RG))
```

Comment:
I feel like we need a test to verify that `RG_LINE` exists in `BASE_INSTRUCTIONS`. That way, if we update `prompt.md`, we are careful to update this function, as well.

Reply:
Also, there appear to be backslashes in front of the backticks in `prompt.md`, so are we sure this matches?